test/integration/VirtMemTests.cc

```diff
@@ -27,6 +27,9 @@ TEST(VirtMemTest, MmapSysCallNoAddressNoFile) {
   // Create the instance of the OS
   simeng::OS::SimOS simOS = simeng::OS::SimOS(DEFAULT_STR, {}, memory, false);
 
+  // Inject fake syscall into queue
+  simOS.getSyscallHandler()->receiveSyscall({});
```

Comment:
Why do we need to inject a fake syscall? 

Also, if `getSyscallHandler()` is only used by the test suite, is there an alternative way to achieve the same logic?

Reply:
Could we not construct a `SyscallInfo` and use `recieveSyscall` which would pass it through and add it to the queue?

Reply:
That's what is happening at the moment

Reply:
Sorry, the `recieveSyscall` defined in SimOS. Although I'm not sure how you would get the result out if you didn't have direct access to the syscallHandler

Reply:
If we had the test able to access private variables of the SimOS (friend class), then we could inject a fake syscall into the `syscallQueue_` directly and then call `mmap` or `munmap` directly too, I think?

Reply:
Yeah we could, although as VirtMemTest is not a class I'm not sure how to do this nicely. Lets keep it as is for now, and we can have a thing in the planned refactor work that will occur